worker.go

```diff
@@ -641,3 +642,38 @@ func getCloudWatchAlarmsFromConfigMap(configMap *kubernetes.ConfigMap) aws.Cloud
 
 	return configList
 }
+
+// cniEventHandler syncronizes the events from kubernetes and the status updates from the load balancer controller.
+// Events from kubernetes pod updates trigger a rate limited update of the target groups.
+func cniEventHandler(ctx context.Context, targetCNIcfg *aws.TargetCNIconfig,
+	targetSetter func([]string, []string) error, informer func(context.Context, chan<- []string) error) {
+	var cniTargetGroupARNs []string
+	log.Infoln("Starting CNI event handler")
+
+	rateLimiter := time.NewTicker(cniEventRateLimit)
+	defer rateLimiter.Stop()
+
+	epCh := make(chan []string, 10)
+	go informer(ctx, epCh)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case cniTargetGroupARNs = <-targetCNIcfg.TargetGroupCh:
+			log.Debugf("new message target groups: %v", cniTargetGroupARNs)
+		case eps := <-epCh:
+			log.Debugf("new message endpoints: %v", eps)
+			if len(epCh) > 0 {
+				log.Debugf("skipping, %d newer messages queued", len(epCh))
+				continue
+			}
+			// simple rate limit
+			<-rateLimiter.C
+			err := targetSetter(eps, cniTargetGroupARNs)
+			if err != nil {
+				log.Error(err)
+			}
+		}
+	}
```

Comment:
Maybe this should run `targetSetter` on either update of both `TargetGroupCh` and `epCh` like:
```go
	var cniTargetGroupARNs, eps []string
	for {
		select {
		case <-ctx.Done():
			return
		case cniTargetGroupARNs = <-targetCNIcfg.TargetGroupCh:
			log.Debugf("new message target groups: %v", cniTargetGroupARNs)
		case eps = <-epCh:
			log.Debugf("new message endpoints: %v", eps)
		}
		
		if len(epCh) > 0 || len(targetCNIcfg.TargetGroupCh) > 0 {
			continue
		}

		<-rateLimiter.C
		err := targetSetter(eps, cniTargetGroupARNs)
		if err != nil {
			log.Error(err)
		}
	}
```

Reply:
Guess we need to prevent no-op triggers from TargetGroupCh though

Reply:
Current implementation only updates when it receives endpoints update but not target groups update.
I guess this is the reason to have a "resyncer" that constantly sends endpoint updates https://github.com/zalando-incubator/kube-ingress-aws-controller/pull/461#discussion_r787656253

> Guess we need to prevent no-diff triggers from TargetGroupCh though

And from endpoint updates send by resyncer (IMO its better to remove the resyncer altogether by running setter on either update)

Reply:
The purpose of the resync is to imply desired state, against user caused divergence, not internal inconsistency.
Eventually it doesn't matter which timer source retriggers the sync since it's idempotent, sounds good

Reply:
done